tests/framework/installer/install_helper.go

```diff
@@ -222,61 +217,62 @@ func (h *InstallHelper) UninstallRookFromK8s(clusterNamespace string, helmInstal
 	logger.Infof("Uninstalling Rook")
 	k8sHelp, err := utils.CreateK8sHelper(h.T)
 	if err != nil {
-		panic(err)
+		logger.Errorf("cannot uninstall rook err ->  %v", err)
 	}
 	if helmInstalled {
 		err = h.helmHelper.DeleteLocalRookHelmChart(helmDeployName)
 		if err != nil {
-			panic(err)
+			logger.Errorf("cannot uninstall Rook helm chart err -> %v", err)
 		}
 	} else {
-		rookOperator := h.installData.GetRookOperator(h.k8shelper.GetK8sServerVersion())
+		rookOperator := h.installData.GetRookOperator()
 
 		_, err = h.k8shelper.KubectlWithStdin(rookOperator, deleteArgs...)
 		if err != nil {
-			panic(err)
+			logger.Errorf("cannot uninstall rook-operator err -> %v", err)
 		}
-		k8sHelp.WaitUntilNameSpaceIsDeleted("rook-system")
 	}
 	_, err = k8sHelp.DeleteResource([]string{"-n", clusterNamespace, "cluster", clusterNamespace})
 	if err != nil {
-		panic(err)
+		logger.Errorf("cannot remove cluster %s in namespace %s  err -> %v", clusterNamespace, clusterNamespace, err)
 	}
 	_, err = k8sHelp.DeleteResource([]string{"-n", clusterNamespace, "serviceaccount", "rook-api"})
 	if err != nil {
-		panic(err)
+		logger.Errorf("cannot remove serviceaccount rook-api in namespace %s  err -> %v", clusterNamespace, err)
 	}
 	_, err = k8sHelp.DeleteResource([]string{"-n", clusterNamespace, "serviceaccount", "rook-ceph-osd"})
 	if err != nil {
-		panic(err)
+		logger.Errorf("cannot remove serviceaccount rook-ceph-osd in namespace %s  err -> %v", clusterNamespace, err)
 	}
 	if h.k8shelper.VersionAtLeast("v1.6.0") {
 		err = k8sHelp.DeleteClusterRoleAndBindings("rook-api")
 		if err != nil {
 			logger.Errorf("rook-api cluster role and binding cannot be deleted: %+v", err)
-			panic(err)
 		}
 		err = k8sHelp.DeleteClusterRoleAndBindings("rook-ceph-osd")
 		if err != nil {
 			logger.Errorf("rook-ceph-osd cluster role and binding cannot be deleted: %+v", err)
-			panic(err)
 		}
 	}
 
 	if h.k8shelper.VersionAtLeast("v1.7.0") {
 		_, err = k8sHelp.DeleteResource([]string{"crd", "clusters.rook.io", "pools.rook.io", "objectstores.rook.io", "filesystems.rook.io"})
 		if err != nil {
-			panic(err)
+			logger.Errorf("cannot delete CRDs err -> %+v", err)
 		}
 	} else {
 		_, err = k8sHelp.DeleteResource([]string{"thirdpartyresources", "cluster.rook.io", "pool.rook.io", "objectstore.rook.io", "filesystem.rook.io"})
 		if err != nil {
-			panic(err)
+			logger.Errorf("cannot delete TPRs err -> %+v", err)
 		}
 	}
+	_, err = k8sHelp.DeleteResource([]string{"secret", clusterNamespace + "-rook-user"})
```

Comment:
do we need to delete the namespace-specific resources since the namespace will clean that up when it's deleted?

Reply:
There clean up steps reflect documentaion, I can remove them if we think they are redundant.

Reply:
if the user chooses not to delete the entire namespace, then the individual cleanup instructions would be necessary. But since the test is deleting the namespace, it could skip deleting the secret and other artifacts in the namespace. You can save this for another PR though